bgpd/bgp_clist.c

```diff
@@ -666,8 +667,14 @@ bool community_list_match(struct community *com, struct community_list *list)
 			if (community_match(com, entry->u.com))
 				return entry->direct == COMMUNITY_PERMIT;
 		} else if (entry->style == COMMUNITY_LIST_EXPANDED) {
-			if (community_regexp_match(com, entry->reg))
-				return entry->direct == COMMUNITY_PERMIT;
+			if (!com)
```

Comment:
This seems can be checked at the beginning of the function to avoid calling it multiple times?

Reply:
the implementation changed. pls reconsider it